File: lista 1 (funções)/questao4.cpp
#include <iostream>
#include <cassert>

using namespace std;

int somaNaturais(int numero_natural){
    int soma = 0;

    for(int i = 0; i <= numero_natural; i++){  

        soma = soma + i;
   }

   return soma;
}

int main() {
    // Testes unitários
    assert(somaNaturais(5) == 15);  // 1 + 2 + 3 + 4 + 5 = 15
    assert(somaNaturais(1) == 1);   // 1 = 1
    assert(somaNaturais(10) == 55); // 1 + 2 + ... + 10 = 55
    assert(somaNaturais(0) == 0);   // Caso com n = 0
    assert(somaNaturais(3) == 6);   // 1 + 2 + 3 = 6

    cout << "Todos os testes passaram!" << endl;

    return 0;
}

File: lista 1 (funções) rust/questao4.rs
/*

Crie uma função chamada somaNaturais que receba um número inteiro positivo n como parâmetro
 e retorne a soma de todos os números naturais de 1 até n. 
 A função deve utilizar a estrutura de repetição for para calcular a soma.

*/

fn soma_naturais(n: u32) -> u32 {
    let mut soma_acumulada: u32 = 0;
    for i in 1..(n + 1) {
        soma_acumulada += i;
    }
    soma_acumulada
}

fn main() {
    assert!(soma_naturais(5) == 15); // 1 + 2 + 3 + 4 + 5 = 15
    assert!(soma_naturais(1) == 1); // 1 = 1
    assert!(soma_naturais(10) == 55); // 1 + 2 + ... + 10 = 55
    assert!(soma_naturais(0) == 0); // Caso com n = 0
    assert!(soma_naturais(3) == 6); // 1 + 2 + 3 = 6

    println!("Todos os testes passaram!");
}
